components/Tabs/tests/snapshot/Theming/MDCTabBar_MaterialThemingSnapshotTests.m

```diff
@@ -0,0 +1,130 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialSnapshot.h"
+
+#import "MaterialContainerScheme.h"
+#import "MaterialTabs+Theming.h"
+#import "MaterialTabs.h"
+
+static NSString *const kItemTitle1 = @"Item 1";
+static NSString *const kItemTitle2 = @"Item 2";
+static NSString *const kItemTitle3 = @"Item 3";
+static NSString *const kItemBadgeValue = @"Badge";
+
+/** Snapshot tests for MDCTabBar rendering .*/
+@interface MDCTabBar_MaterialThemingSnapshotTests : MDCSnapshotTestCase
+
+/** The view being tested. */
+@property(nonatomic, strong) MDCTabBar *tabBar;
+@property(nonatomic, strong) UITabBarItem *item1;
+@property(nonatomic, strong) UITabBarItem *item2;
+@property(nonatomic, strong) UITabBarItem *item3;
+@property(nonatomic, strong) MDCContainerScheme *containerScheme;
+@end
+
+@implementation MDCTabBar_MaterialThemingSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Uncomment below to recreate all the goldens (or add the following line to the specific
+  // test you wish to recreate the golden for).
+  // self.recordMode = YES;
+
+  CGSize imageSize = CGSizeMake(24, 24);
+  self.tabBar = [[MDCTabBar alloc] init];
+  self.item1 = [[UITabBarItem alloc]
+      initWithTitle:kItemTitle1
+              image:[[UIImage mdc_testImageOfSize:imageSize
+                                        withStyle:MDCSnapshotTestImageStyleCheckerboard]
+                        imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]
+                tag:1];
+  self.item2 = [[UITabBarItem alloc]
+      initWithTitle:kItemTitle2
+              image:[[UIImage mdc_testImageOfSize:imageSize
+                                        withStyle:MDCSnapshotTestImageStyleDiagonalLines]
+                        imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]
+                tag:2];
+  self.item3 = [[UITabBarItem alloc]
+      initWithTitle:kItemTitle3
+              image:[[UIImage mdc_testImageOfSize:imageSize
+                                        withStyle:MDCSnapshotTestImageStyleFramedX]
+                        imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]
+                tag:3];
+  self.item3.badgeValue = kItemBadgeValue;
+  self.tabBar.items = @[ self.item1, self.item2, self.item3 ];
+
+  self.containerScheme = [[MDCContainerScheme alloc] init];
+}
+
+- (void)tearDown {
+  self.tabBar = nil;
+  self.item1 = nil;
+  self.item2 = nil;
+  self.item3 = nil;
+
```

Comment:
Nit: Nil-out the container scheme.

Reply:
Addressed in 4e8cb3cc3ec6eb9a0b5536dac519b6b866791415